libs/ngx-dsfr/upload/src/lib/testing/upload.component.spec.ts

```diff
@@ -0,0 +1,103 @@
+/**
+ * Angular imports
+ */
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { HarnessLoader } from '@angular/cdk/testing';
+import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
+import { FormsModule } from '@angular/forms';
+
+/**
+ * 3rd-party imports
+ */
+import { RouterLinkDirectiveStub } from '@betagouv/ngx-dsfr/testing';
+
+/**
+ * Internal imports
+ */
+import { TestHostComponent } from './test-host.component';
+import { DsfrUploadComponent, EMPTY_ID_ERROR, EMPTY_NAME_ERROR } from '../upload.component';
+import { DsfrUploadHarness } from './upload.harness';
+
+describe('DsfrFooterComponent', () => {
+  let fixture: ComponentFixture<TestHostComponent>;
+  let testHost: TestHostComponent;
+  let componentUnderTest: DsfrUploadComponent;
+  let harnessLoader: HarnessLoader;
+  let dsfrUploadHarness: DsfrUploadHarness;
+
+  const testName = 'testName';
+  const testLabel = 'testLabel';
+  const testId = 'testId'
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [FormsModule],
+      declarations: [
+        DsfrUploadComponent,
+        TestHostComponent,
+        RouterLinkDirectiveStub
+      ]
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(TestHostComponent);
+    testHost = fixture.componentInstance;
+    componentUnderTest = fixture.debugElement.query(
+      By.directive(DsfrUploadComponent)
+    ).componentInstance;
+
+    /*
+     * We're retrieving a HarnessLoader HERE to be able to get
+     * the Test Harness for the DsfrUploadComponent LATER.
+     *
+     * We're NOT doing both HERE since getHarness() triggers
+     * indirectly the Change Detection mechanism and this,
+     * Data Binding and ngOnInit()...
+     */
+    harnessLoader = TestbedHarnessEnvironment.loader(fixture);
+  });
+
+  it('should be created', () => {
+    expect(componentUnderTest).toBeTruthy();
+  });
+
+  it('should throw an error when no id is provided', async () => {
```

Comment:
```typescript
it('should throw an error when no hint is provided', async () => {
    try {
      fixture.detectChanges();
      throw 'It should have thrown an error about "HINT"';
    } catch (error) {
      expect(error).toBe(EMPTY_HINT_ERROR);
    }
  });
```

It needs to come first because we put the check on `hint` inside `ngOnChanges`, which is called for the first time before `ngOnInit`...